### PR TITLE
Binary focal loss: Use pre-computed probs and increase readability

### PR DESCRIPTION
**Pull request by klieret**
#### Changes

* Use pre-computed probs_pos and probs_neg rather than calling sigmoid function again (performance boost by skipping two calls to `torch.sigmoid`)
* Standardize use of tensor method vs torch function to increase readability (make symmetry in code more apparent)

#### Type of change

Performance / code readability

- [ ] 📚  Documentation Update
- [ ] 🧪 Tests Cases
- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [ ] 🔬 New feature (non-breaking change which adds functionality)
- [ ] 🚨 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📝 This change requires a documentation update

#### Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Did you update CHANGELOG in case of a major change?

### CODE REVIEW

File: kornia/losses/focal.py [hunk 1]
```diff
@@ -209,12 +209,12 @@ def binary_focal_loss_with_logits(
     KORNIA_CHECK_IS_TENSOR(pos_weight)
     KORNIA_CHECK(input.shape[-1] == pos_weight.shape[0], "Expected pos_weight equals number of classes.")
 
-    probs_pos = input.sigmoid()
+    probs_pos = torch.sigmoid(input)
```
Comment (edgarriba) [inline kornia/losses/focal.py:212]: This one was made in purpose towards to "normalise" the library to not only pytorch

**Review by edgarriba — APPROVED** [review]

File: kornia/losses/focal.py (hunk 1 above)
Comment (klieret) [reply kornia/losses/focal.py:212]: Thank you for reviewing the changes. Ah, you're saying that `x.sigmoid()` "looks" more "library-independent" and was preferred? I could change the second line to `(-input).sigmoid()` instead and roll back this one if you want...

Comment (edgarriba) [reply kornia/losses/focal.py:212]: Yeah, that’s right. All rendir libs are converging apis :)

Comment (klieret) [reply kornia/losses/focal.py]: Thanks. I switched it over to use the method now :)
